class.photon.php

```diff
@@ -65,6 +65,161 @@ private function setup() {
+		add_filter( 'intermediate_image_sizes_advanced', array( __CLASS__, 'filter_photon_noresize_intermediate_sizes' ) );
+
+		// Load the noresize srcset solution on priority of 20, allowing other plugins to set sizes earlier.
+		add_filter( 'wp_get_attachment_metadata', array( __CLASS__, 'filter_photon_norezise_maybe_inject_sizes' ), 20, 2 );
+
+		// Photonize thumbnail URLs in the API response.
+		add_filter( 'rest_api_thumbnail_size_urls', array( __CLASS__, 'filter_photon_noresize_thumbnail_urls' ) );
+
+		// This allows to assign the Photon domain to images that normally use the home URL as base.
+		add_filter( 'jetpack_photon_domain', array( __CLASS__, 'filter_photon_norezise_domain' ), 10, 2 );
+
+		add_filter( 'the_content', array( __CLASS__, 'filter_content_add' ), 0 );
+
+		// Jetpack hooks in at six nines (999999) so this filter does at seven.
+		add_filter( 'the_content', array( __CLASS__, 'filter_content_remove' ), 9999999 );
+
+		// Regular Photon operation mode filter doesn't run when is_admin(), so we need an additional filter.
+		// This is temporary until Jetpack allows more easily running these filters for is_admin().
+		if ( is_admin() ) {
+			add_filter( 'image_downsize', array( $this, 'filter_image_downsize' ), 5, 3 );
+
+			// Allows any image that gets passed to Photon to be resized via Photon.
+			add_filter( 'jetpack_photon_admin_allow_image_downsize', '__return_true' );
+		}
+	}
+
+	/**
+	 * This is our catch-all to strip dimensions from intermediate images in content.
+	 * Since this primarily only impacts post_content we do a little dance to add the filter early
+	 * to `the_content` and then remove it later on in the same hook.
+	 *
+	 * @param String $content the post content.
+	 * @return String the post content unchanged.
+	 */
+	public static function filter_content_add( $content ) {
+		add_filter( 'jetpack_photon_pre_image_url', array( __CLASS__, 'strip_image_dimensions_maybe' ) );
+		return $content;
+	}
+
+	/**
+	 * Removing the content filter that was set previously.
+	 *
+	 * @param String $content the post content.
+	 * @return String the post content unchanged.
+	 */
+	public static function filter_content_remove( $content ) {
+		remove_filter( 'jetpack_photon_pre_image_url', array( __CLASS__, 'strip_image_dimensions_maybe' ) );
+		return $content;
+	}
+
+	/**
+	 * Short circuits the Photon filter to enable Photon processing for any URL.
+	 *
+	 * @param String $photon_url a proposed Photon URL for the media file.
+	 * @param String $image_url the original media URL.
+	 * @return String an URL to be used for the media file.
+	 */
+	public static function filter_photon_norezise_domain( $photon_url, $image_url ) {
+		return $photon_url;
+	}
+
+	/**
+	 * Disables intermediate sizes to disallow resizing.
+	 *
+	 * @param Array $sizes an array containing image sizes.
+	 * @return Boolean
+	 */
+	public static function filter_photon_noresize_intermediate_sizes( $sizes ) {
+		return array();
```

Comment:
@jeherve I had to bring back this method because it's getting attached and removed at runtime in the REST API filter. I didn't feel like leaving `__return_empty_array` here because we might be removing someone else's filter too.